packages/manager/modules/pci/src/projects/project/instances/add/legacy/add.controller.js

```diff
@@ -157,6 +158,37 @@ export default class PciInstancesAddController {
     this.getFilteredRegions();
   }
 
+  displayFlavorSelectionHeader() {
+    if (this.currentStep > 0) {
+      // if the flavor doesn't have a monthly plan
+      if (this.model.flavorGroup?.tagsBlob?.includes(TAGS_BLOB.COMING_SOON)) {
+        return this.$translate.instant(
+          'pci_projects_project_instances_add_flavor_selected_title_witout_price',
+          {
+            model: this.model?.flavorGroup?.name.toUpperCase(),
+          },
+        );
+      }
+
+      // if the flavor does have a monthly plan
+      return this.$translate.instant(
+        'pci_projects_project_instances_add_flavor_selected_title',
+        {
+          model: this.model?.flavorGroup?.name.toUpperCase(),
+          price: this.model?.flavorGroup?.prices?.monthly?.text,
+        },
+      );
+    }
+    // current step is the flavor selection step
+    return this.$translate.instant(
+      'pci_projects_project_instances_add_flavor_title',
+    );
+  }
+
+  IsComingSoonPricingBannerDisplayed() {
+    return !!this.model.flavorGroup?.tagsBlob?.includes(TAGS_BLOB.COMING_SOON);
```

Comment:
**nitpick:** Unnecessary `!!` since the `includes` method returns the boolean.
```suggestion
    return this.model.flavorGroup?.tagsBlob?.includes(TAGS_BLOB.COMING_SOON);
```